NExtends.Tests/Helpers/JsonExtensions.cs

```diff
@@ -0,0 +1,198 @@
+                result = DateTime.ParseExact(date, "ddd MMM d HH:mm:ss UTCzzzzz yyyy", CultureInfo.InvariantCulture);
+            }
+
+            return result;
+        }
+
+
+        public static String JSONProtect(this String value)
+        {
+            if (value != null)
+            {
+                String json = value.Replace("\r\n", "<br>");
+                json = json.Replace("\n", "<br>");
+                json = json.Replace("\r", "<br>");
+
+                json = json.Replace("\\", "\\\\");
+
+                json = json.Replace("'", "\\'");
+                json = json.Replace("\"", "\\\"");
+
+                return json;
+            }
+            else
+            {
+                return "";
+            }
+        }
+
+
+        //Renvoie une date au format JSON, limite à 2099 pour les MaxValue
+
+        public static String ToJSONDate(this DateTime d)
+        {
+            return "new Date(" + Math.Min(d.Year, 2099) + ", " + (d.Month - 1) + ", " + d.Day + ")";
+        }
+
+        public static String ToJSONDateTime(this DateTime d)
+        {
+            return "new Date(" + Math.Min(d.Year, 2099) + ", " + (d.Month - 1) + ", " + d.Day + ", " + d.Hour + ", " + d.Minute + ", " + d.Second + ")";
+        }
+
+        public static String ToJQuery(this DateTime d)
+        {
+            return JsonConvert.SerializeObject(d);
+        }
+
+        public static String ToJQuery(this DateTime? d)
+        {
+            return JsonConvert.SerializeObject(d);
+        }
+
+        [Obsolete("Use ToISO instead and parse the date on the client side")]
+        public static String ToJSON(this DateTime d)
+        {
+            return "new Date(" + Math.Min(d.Year, 2099) + ", " + (d.Month - 1) + ", " + d.Day + ", " + d.Hour + ", " + d.Minute + ", " + d.Second + ")";
+        }
+
+        public static String ToJSON(this Double d)
+        {
+            return d.ToString(CultureInfo.InvariantCulture);
+        }
+
+        /// <summary>
+        /// Converti un enum {Option1 = 1, Option2 = 2} en JSON {Option1: 1, Option2: 2}
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <returns></returns>
+        public static String EnumToJSON<T>() where T : struct
+        {
+            var JSONProperty = new Func<T, string>(e =>
+            {
+                return String.Format("{0}:{1}",
+                    e.ToString().ToDoubleQuotesJSON(),
+                    Enum.Format(typeof(T), e, "d")
+                );
+            });
+
+            return "{" + String.Join(",", EnumExtensions.EnumToList<T>().Select(JSONProperty).ToArray()) + "}";
+        }
+
+        public static String ToJSON<T>(this IEnumerable<T> list)
+        {
+            return "[" + String.Join(",", list.Select(o => (string)o.GetType().GetMethod("ToJSON", new Type[] { }).Invoke(o, null)).ToArray()) + "]";
+        }
+    }
+}
```

Comment:
From my point of view, we don't need all theses methods anymore, since we only use them in legacy code in Lucca. And for me the "best practice" with serialization would be :
- use Json.NET instead of home made extensions
- extend the json serialization on some type (Enum & co) inside the web layer using Json.NET overridable serializers